include/seastar/core/shared_ptr.hh

```diff
@@ -738,35 +766,35 @@ SEASTAR_MODULE_EXPORT_BEGIN
 template <typename T, typename U>
 inline
 bool
-operator==(const shared_ptr<T>& x, const shared_ptr<U>& y) {
+operator==(const shared_ptr<T>& x [[clang::param_typestate(unconsumed)]], const shared_ptr<U>& y [[clang::param_typestate(unconsumed)]]) {
```

Comment:
Are these annotations needed or helpful? This function calls x.get() on x, so should already fail if x is consumed. Or maybe the "unknown" case is the difference here?

Reply:
i don't think so. dropped in the latest revision. actually. i don't think we should enforce any expectation here. as any `shared_ptr` can be compared with any `shared_ptr`. and neither should we enforce any expectation when calling `get()`, as `shared_ptr::get()` should behave no matter it is consumed or not.